OperationCode/resources_api
# Change blacklisted -> denied for inclusivity

The word `blacklist` isn't a very inclusive term. I think that `denylist` is a much better term to use, and besides, it's more descriptive for what we're doing to the API key.

This PR involves a DB migration and also has everything to do with our authentication, so please give it a solid review. To run the migration locally, you should be able to run `make migrate`. Let me know if this doesn't work, because we'll want to fix that command. And of course, let me know if you run into any issues.

Testing:
1) Get your API key with the `/apikey` endpoint
2) Use the API key to successfully add a resource with `POST /resources`
3) Use the API key to successfully update a resource with `PUT /resources`
4) Deny your API key using `docker-compose run resources-api flask apikey deny <email address>` (can also pass in API key instead of email address)
5) Ensure you can't get your API key from the `/apikey` endpoint after it's been put on the denylist
6) Ensure you can't make a `POST` or `PUT` request to the `/resources` endpoint with your denied API key
7) Reactivate your API key using `docker-compose run resources-api flask apikey reactivate <email address>` (can also pass in API key instead of email address)
8) Get your API key again and make sure you're allowed to make POST requests again.
